addons/Hierarchical_Lesson_Report/src/presenter.js

```diff
@@ -1253,6 +1267,12 @@ function AddonHierarchical_Lesson_Report_create() {
         checkIfChapterHasChildren();
     };
 
+    presenter.setPagesVisited = function () {
```

Comment:
Addon ustawia, że wszystkie strony zostały odwiedzone. Zmiana ta wpłynie na inne addony (najprawdopodobniej tylko te od punktacji) jednak też będzie wpływać na punktację w lekcji. mC korzysta z PlayerUtils w celu wyliczenia punktów. W PlayerUtils w getPresentationScoreBase dla stron raportowalnych dla pewnej ścieżki wpływa to na zmianę punktacji w lekcji. 

Zgodnie z opisem w tickecie addon ma mieć zmienione obecne zachowanie dlatego mam duże obawy, że wystąpią przez to problemy. W opisie zadania jest napisane o zmianie punktacji "jakby" wszystkie strony były odwiedzone. Natomiast obecne rozwiązanie sprawi, że wszystkie strony "zostaną" odwiedzone. 

Dodatkową obawą jest dla mnie działanie Adaptive Learning - Działa to tak, że daje się warunki aby wybrać strony lekcji na które można wejść. Możliwe, że istnieje content, który sprawdza czy jakaś strona została odwiedzona lub jest przyznawany punkt tylko za odwiedzenie strony (bo można wejść tylko na jedną z dwóch stron).

Prywatnie pisaliśmy o tymczasowej zmianie stanu odwiedzenia tych stron. Niestety mam wrażenie, że takie rozwiązanie także naraża na błędy. W mC można ustawić, aby co kilka sekund był zapisywany wynik (automatyczny zapis). Jeżeli wykonałby się w takcie tej operacji to powstaną nieprawidłowości.
Widzę, że w lekcji korzysta się z presenter.isPageVisited. W pierwszej kolejności proszę sprawdź czy nie wystarczy tam zwracać true jeżeli nowe property jest odznaczone.